pkg/loop/internal/plugin_service.go

```diff
@@ -212,11 +209,47 @@ func (s *pluginService[P, S]) closeClient() (err error) {
 	return
 }
 
-func (s *pluginService[P, S]) wait(ctx context.Context) error {
+func (s *PluginService[P, S]) WaitCtx(ctx context.Context) error {
 	select {
 	case <-ctx.Done():
 		return context.Cause(ctx)
 	case <-s.serviceCh:
 		return nil
+	case <-s.stopCh:
+		return errors.New("service was stopped while waiting")
+	}
+}
+
+func (s *PluginService[P, S]) Wait() error {
```

Comment:
can you add a comment above what is being waited for?

when i read the use in NewReportingPlugin() it's not clear why a Wait is needed and in isolation, I would have expected a call to something like Start() or WaitForStart(); `Wait` seems to nebulous without a good comment